modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankedListQualityMetric.java

```diff
@@ -36,6 +44,27 @@
      * @param hits the result hits as returned by some search
      * @return some metric representing the quality of the result hit list wrt. to relevant doc ids.
      * */
-    @Override
-    EvalQueryQuality evaluate(SearchHit[] hits, RatedQuery intent);
+    public abstract EvalQueryQuality evaluate(SearchHit[] hits, List<RatedDocument> ratedDocs);
+
+    public static RankedListQualityMetric fromXContent(XContentParser parser, ParseFieldMatcherSupplier context) throws IOException {
+        RankedListQualityMetric rc;
+        Token token = parser.nextToken();
+        if (token != XContentParser.Token.FIELD_NAME) {
+            throw new ParsingException(parser.getTokenLocation(), "[_na] missing required metric name");
+        }
+        String metricName = parser.currentName();
+
+        switch (metricName) {
```

Comment:
I think this is fine for now. While I do think expert users will want to tweak exactly which metric they use for evaluation, I guess it's ok to not make this plug-able right from the start.